src/model/formats/abstract-format.ts

```diff
@@ -7,53 +7,39 @@ export function getFileExtension(fileName: string): FileExtension {
   return fileName.slice(fileName.lastIndexOf('.') + 1).toLowerCase();
 }
 
-export abstract class FormatOptions {
-  protected constructor(
-    public readonly description: string,
-    public readonly supportsImport: boolean,
-    public readonly extension?: FileExtension,
-  ) {}
+export interface FormatOptions {
+  supportsImport?: boolean;
+  extension?: FileExtension;
 }
 
-export class DefaultFormatOptions extends FormatOptions {
-  constructor(
-    public override readonly description: string,
-    // eslint-disable-next-line @typescript-eslint/no-inferrable-types
-    public override readonly supportsImport: boolean = true,
-    public override readonly extension?: FileExtension,
-  ) {
-    super(description, supportsImport, extension);
-  }
-}
+export type FormatConstructor<T extends FormatOptions> = new (
+  formatId: FormatId,
+  name: string,
+  args?: FormatOptions,
+) => AbstractChecklistFormat<T>;
+
+export abstract class AbstractChecklistFormat<T extends FormatOptions = FormatOptions> {
+  public readonly supportsImport: boolean;
+  protected readonly _extension?: FileExtension;
 
-export abstract class AbstractChecklistFormat<T extends FormatOptions = DefaultFormatOptions> {
   constructor(
     private readonly _formatId: FormatId,
-    protected readonly _formatOptions: T,
-  ) {}
-
-  public get description(): string {
-    return this._formatOptions.description;
-  }
-
-  public get supportsImport(): boolean {
-    return this._formatOptions.supportsImport;
+    public readonly name: string,
+    args?: T,
+  ) {
+    this.supportsImport = args?.supportsImport ?? true;
+    this._extension = args?.extension;
   }
 
   public get extension(): FileExtension {
-    return this._formatOptions.extension ?? this._formatId;
+    return this._extension ?? this._formatId;
```

Comment:
Won't `_formatId` come without the dot that's required for FileExtension? (why does Typescript let you convert a FormatId to a FileExtension like that?)

Reply:
Just realized that you changed this in a later commit.